Commit 1

Gomina opened the pull request:
Добавлены тесты и отчет pytest-cov

Gomina committed:
Commit 1
Добавлены тесты и отчет pytest-cov

DolgoarshinnykhLG commented on conftest.py:
```diff
@@ -0,0 +1,37 @@
+# метод создает и возвращает новый экземпляр класса Burger
+@pytest.fixture
+def burger():
+    return Burger()
```
Нужно исправить: фикстуры не занимаются выполнением примитивной логики, они выполняют сложную логику предусловий\постусловий и вычислений. Эти объекты можно прямо создавать в тестах или фикстурах

DolgoarshinnykhLG commented on tests/test_bun.py:
```diff
@@ -0,0 +1,9 @@
+    def test_new_bun(self):
+        bun = Bun("Булка смерти", 405)
+        assert bun.get_name() == "Булка смерти"
+        assert bun.get_price() == 405
```
Нужно исправить: стоит разделить эти тесты, юнит-тесты должны быть максимально атомарными

DolgoarshinnykhLG commented on tests/test_ingredient.py:
```diff
@@ -0,0 +1,20 @@
+class TestIngredient:
+
+    @pytest.mark.parametrize("ingredient_type, name, price", [
+    (INGREDIENT_TYPE_SAUCE,"Плазменный нектар", 349),  # ингредиент соус
+    (INGREDIENT_TYPE_FILLING, "Антивещество", 567)  # ингредиент начинка
```
Можно лучше здесь и далее: тестовые данные и ожидаемые ответы лучше не хранить в теле/модулях теста. Лучше вынести их в модуль data, так их будет легче поддерживать

Gomina committed:
Commit 2
Убрана фикстура, создающая бургер
@pytest.fixture
def burger():

Gomina committed:
Commit 3
Убрана фикстура, создающая базу
@pytest.fixture
def database():

Gomina committed:
Commit 4
Тест  test_new_bun разделен на два теста: проверяющий нзвание новой булки и проверяющий её стоимость.

Gomina committed:
Commit 5
1. Убраны фикстуры, не занимающиеся сложной логикой: предусловий\постусловий и вычислений.
2. Тестовые данные внесены в модуль data
3. После внесенных обновлений обновлен отчет pytest-cov

DolgoarshinnykhLG commented on conftest.py:
```diff
@@ -0,0 +1,8 @@
+from praktikum.ingredient import Ingredient
+from praktikum.ingredient_types import INGREDIENT_TYPE_SAUCE, INGREDIENT_TYPE_FILLING
+from praktikum.database import Database
+
```
От неиспользуемого модуля можно сразу избавиться

Gomina committed:
Commit 6
Файл conflicts отчищен. Вся неиспользуемая информация удалена.
